integration/write_pyarrow.py

```diff
@@ -85,3 +89,9 @@ def write_pyarrow(case, size = 1, page_version = 1):
 write_pyarrow(case_basic_required, 1, 2)  # V2
 
 write_pyarrow(case_nested, 1, 1)
+
+# pyarrow seems to write corrupt file when disabling dictionary encoding for nullable strings
```

Comment:
@jorgecarleitao I added tests but seem to have run into a problem with pyarrow itself. Have you seen something like this before? For now I have ignored the corresponding test.